### DISCUSSION TITLE
How do I edit profile info

### DISCUSSION THREAD

**Discussion by TxzK** [neet/masto.js#1031] [Q&A] [answered]
How do edit profile information such as username or fields and such? I might be just stupid but I can't seem to find anything. Or is that not possible for some reason?

**Comment by neet** [comment, marked as answer]
Hi @alzawad26, thank you for posting a question. You can edit a profile for an authorised user by `rest.v1.accounts.updateCredentials` (Indeed it has a little bit tricky name). 

You can use it as follows:

```ts
import { createRestAPIClient } from "masto";

const client = createRestAPIClient({ ... });

await client.v1.accounts.updateCredentials({
  displayName: "My new name",
  fieldsAttributes: [
    {
      name: "my title",
      value: "my value",
    },
  ],
});
```

You can see other available parameters on the official documentation. I hope this helps
https://docs.joinmastodon.org/methods/accounts/#update_credentials

**Reply by TxzK** [reply]
Oh thanks! Guess I didn't look hard enough 😅
